thavalon-webapp/src/components/Register.tsx

```diff
@@ -1,32 +1,92 @@
-import React, {useEffect, useState} from 'react';
+import React, {useState} from 'react';
 import ReactModal from 'react-modal';
 import { useForm } from 'react-hook-form';
 import "./modal.scss";
-import { register_user } from '../utils/account_utils';
+import { AccountManager, HttpResponse } from '../utils/accountManager';
+import { Redirect } from 'react-router-dom';
+
+type RegisterProps = {
+    setLoggedIn: any
+};
+
+interface RegisterData {
+    "name": string,
+    "email": string,
+    "password": string,
+    "confirmPassword": string,
+};
 
 ReactModal.setAppElement("#root");
 
-function Login() {
+export function Register(props: RegisterProps) {
+    // if set, register modal is open
     const [modalIsOpen, setModalIsOpen] = useState(true);
-    const {register, handleSubmit, errors} = useForm();
+    // hook for register form
+    const {register, handleSubmit, errors} = useForm<RegisterData>();
+    // state for setting if register button is disabled
     const [disable, setDisabled] = useState(false);
+    // state for setting register error
+    const [formErrorMsg, setFormErrorMsg] = useState("");
+    // state for redirecting to home on successful login
+    const [redirectToHome, setRedirectToHome] = useState(false);
+
+    /**
+     * Called when register modal is closed.
+     */
     function closeModal() {
         setModalIsOpen(false);
     }
 
+    /**
+     * On error, just prevent page reload - form handles showing errors.
+     * @param data The data being sent on error.
+     * @param event The event caused by submission.
+     */
     function onError(data: any, event: any) {
-        setDisabled(false);
-        console.log(data);
-        console.log("ERROR");
         event.preventDefault();
     }
 
-    function onSubmit(data: any, event: any) {
-        console.log("registering user with data: " + data);
-        register_user(data.name, data.email, data.password);
+    /**
+     * On error, just prevent page reload - form handles showing errors.
+     * @param data The data being sent on submit.
+     * @param event The event caused by submission.
+     */
+    async function onSubmit(data: RegisterData, event: any) {
+        // disable button on start of submit
+        // TODO: Also add loading image
+        setDisabled(true);
+        setFormErrorMsg("");
+
+        // confirm passwords match. If they do not, show error message.
+        if (data.password !== data.confirmPassword) {
+            setFormErrorMsg("Password do not match");
+            setDisabled(false); 
+            event.preventDefault();
+            return;
+        }

```

Comment:
I think we can build this into the form validation with a resolver: https://github.com/react-hook-form/react-hook-form/blob/master/examples/types/Resolver.tsx